shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java

```diff
@@ -1236,13 +1236,35 @@ public boolean isPathEncrypted(Path path) throws IOException {
       } else {
         fullPath = path.getFileSystem(conf).makeQualified(path);
       }
-      if(!"hdfs".equalsIgnoreCase(path.toUri().getScheme())) {
+      if (!isFileInHdfs(path.getFileSystem(conf), path)) {
         return false;
       }
 
       return (getEncryptionZoneForPath(fullPath) != null);
     }
 
+    /**
+     * Returns true if the given fs supports mount functionality. In general we
+     * can have child file systems only in the case of mount fs like
+     * ViewFileSystem, ViewFsOverloadScheme or ViewDistributedFileSystem.
+     * Returns false if the getChildFileSystems API returns null.
+     */
+    private boolean isMountedFs(FileSystem fs) {
+      return fs.getChildFileSystems() != null;
+    }
+
+    private boolean isFileInHdfs(FileSystem fs, Path path) throws IOException {
+      String hdfsScheme = "hdfs";
+      boolean isHdfs = hdfsScheme.equalsIgnoreCase(path.toUri().getScheme());
```

Comment:
Is there a case that HDFS file system can be mounted as other file systems?

Reply:
@yongzhi Thanks for taking a look. yes, we have implemented ViewDistributedFileSystem and ViewFsOverloadScheme in hadoop. They both supports hdfs paths can be mounted with other fs. You can see in the description and traces that hdfs paths pointing to o3fs paths.

Reply:
Thanks for offline discussion @yongzhi. I have updated comments for better clarity as you suggested!!